docs/source/developers/guide/step_by_step/set_up.rst

```diff
@@ -27,3 +27,98 @@
 ******
 Set up
 ******
+
+Install and setup Git
+=====================
+
+The Arrow project is developed using `git <https://git-scm.com/>`_
+for version control which is easily available for all common
+operating systems.
+
+You can follow the instructions to install git from GitHub
+where Arrow repository is hosted, following
+`the quickstart instructions <https://docs.github.com/en/get-started/quickstart/set-up-git>`_.
+
+When git is set up do not forget to configure your name and email
+
+.. code:: console
+
+   $ git config --global user.name "Your Name"
+   $ git config --global user.email your.email@example.com
+
+and `authenticate with GitHub <https://docs.github.com/en/get-started/quickstart/set-up-git#next-steps-authenticating-with-github-from-git>`_
+as this will allow you to interact with GitHub without typing
+a username and password each time you execute a git command.
+
+**The Terminal**
+
+You might be able to get away without using the command-line for
+most tasks, but for some it seems to be the most straightforward way.
```

Comment:
I'm not sure what the point of this last paragraph is. It seems too vague and uninformative to me. Is there something specific we want to say about the command line here?

Reply:
This was added, if I remember correctly, because not all Arrow users are used to work with the command-line. It serves as a soft warning =) It is not necessary information to have but it is a note to get acquainted with the tool. Was thinking of adding a link with basic terminal commands also.

cc @dragosmg

Reply:
RStudio users can clone a repo directly in RStudio without doing it in the terminal. In the case of arrow that is not advised as it creates additional artefacts (an `.Rproj` file) stemming from the idea that, most frequently and R package sits in the root directory of a repo

Reply:
Then this should be more explicit about it, for example:
```restructuredtext
.. note::
   This guide assumes you are comfortable working from the command line.
   Some IDEs allow you to manage a git repository, but may implicitly run
   unwanted operations when doing so (such as creating project files).
```